src/utils/styles/media.ts

```diff
@@ -8,26 +8,31 @@ export const breakpoints = {
   xlarge: 1275,
 };
 
-type Sizes = keyof typeof breakpoints;
+type Breakpoints = typeof breakpoints;
+type Sizes = keyof Breakpoints;
 
-type MediaQuery = {
-  [key in Sizes]: (cls: any) => string;
+type MediaQuery<T> = {
+  [P in keyof T]: (args: Interpolation, key?: P) => string;
 };
 
-const mq: MediaQuery = Object.keys(breakpoints).reduce(
-  (accumulator, label) => {
-    const prefix = typeof breakpoints[label] === 'string' ? '' : 'min-width:';
-    const suffix = typeof breakpoints[label] === 'string' ? '' : 'px';
-    accumulator[label] = cls =>
-      css`
-        @media (${prefix + breakpoints[label] + suffix}) {
-          ${cls};
-        }
-      `;
-    return accumulator;
-  },
-  // eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
-  {} as MediaQuery
-);
+function constructMQ(breakpoint: Sizes, args: Interpolation): string {
```

Comment:
I see too much unnecessary complexity in this method, I'd go for remove it in the future and use media queries by hand, much easy to understand for new contributors.

Reply:
yeah, I just used the existing code and made it more type safe for the compiler, so it doesn't complain as I didn't want to change the usage in the code.

Reply:
Yes, just was a mental note :-) ... I always wanted to remove it.